client/src/components/order/ordersTable.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from "react";
+import { DataTable } from "primereact/datatable";
+import { Column } from "primereact/column";
+import IOrder from "../../types/IOrder";
+import IItem from "../../types/IItem";
+import { Button } from "primereact/button";
+import OrderPopUp from "./orderPopUp";
+import { ordersStore } from "../../store/ordersStore";
+import { observer } from "mobx-react";
+
+import "primeicons/primeicons.css";
+import "primereact/resources/themes/saga-blue/theme.css";
+import "primereact/resources/primereact.css";
+import "primeflex/primeflex.css";
+import "../../comCss/orders.css";
+
+
+interface IOrderTableProps {
+  orderPopUp: IOrder | undefined;
+  setOrderPopUp: React.Dispatch<React.SetStateAction<IOrder | undefined>>;
+  displayOrderPopUp: boolean;
+  setDisplayOrderPopUp: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const OrdersTable: React.FC<IOrderTableProps> = ({ orderPopUp,setOrderPopUp,displayOrderPopUp,setDisplayOrderPopUp }) => {
+
+  const [selectedOrder, setSelectedOrder] = useState(null);
+
+  const [orderPrice, setOrderPrice] = useState<number>(0);
+
+  const handleOrderDialog = (orderData: IOrder) => {
+    setOrderPopUp({...orderData});
+    setDisplayOrderPopUp((orderD) => !orderD);
+    sumOrderPrice(orderData);
+  }
+
+  const itemsUnits = (orderData: IOrder) => {
+    return <>{orderData.items?.length}</>;
+  };
+
+  const orderDate = (orderData: IOrder) => {
+    if (orderData.date) {
+      const date = new Date(orderData.date.toString());
+      const year = date.getFullYear();
+      const mm = date.getMonth() + 1;
+      const dd = date.getDate() + 2;
+      let day = dd.toString();
+      let month = mm.toString();
+
+      if (dd < 10) day = "0" + dd;
+      if (mm < 10) month = "0" + mm;
+      return (
+        <p>
+          {day}/{month}/{year}
+        </p>
+      );
+    }
```

Comment:
Great! There are libraries that doing that so you don't need to implement it yourself :) find one